src/jit/lsraxarch.cpp

```diff
@@ -36,37 +36,48 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 //
 // Notes:
 // Preconditions:
-//    LSRA Has been initialized and there is a TreeNodeInfo node
-//    already allocated and initialized for every tree in the IR.
+//    LSRA Has been initialized.
+//
 // Postconditions:
-//    Every TreeNodeInfo instance has the right annotations on register
-//    requirements needed by LSRA to build the Interval Table (source,
-//    destination and internal [temp] register counts).
+//    RefPositions have been built for all the register defs and uses required
+//    for this node.
 //
-void LinearScan::BuildNode(GenTree* tree)
+int LinearScan::BuildNode(GenTree* tree)
 {
-    TreeNodeInfo* info = currentNodeInfo;
     assert(!tree->isContained());
-
+    Interval* prefSrcInterval = nullptr;
+    int       srcCount;
+    int       dstCount       = 0;
+    regMaskTP dstCandidates  = RBM_NONE;
+    regMaskTP killMask       = RBM_NONE;
+    pendingDelayFree         = false;
```

Comment:
These member assignments should be visually separated from the local variable above, I'd move them above locals and add a blank line. Or, perhaps they should not even be in `BuildNode` but in its caller.

Also, the need for these data members is a bit unfortunate. It's kind of difficult to track down where they are assigned values and where they are used, the delay free ones in particular.

Reply:
I agree that they are somewhat unfortunate; I tried to come up with an approach that didn't require these data members on `LinearScan`, but could only come up with models that involved excessive passing around of state.
I'd prefer to keep them in `BuildNode`, however, as that's the main entry point for all the building functionality. I'll separate them and document them more clearly.